zend_abstract_interface/properties/php7-8/properties.h

```diff
@@ -0,0 +1,12 @@
+#ifndef ZAI_PROPERTIES_H
+#define ZAI_PROPERTIES_H
+
+#include "Zend/zend_API.h"
+
+// Note that the function will completely bypass read_property. ce (scope) must be object ce or parent of object ce.
+zval *zai_read_property_direct(zend_class_entry *ce, zend_object *object, zend_string *name);
+
+zval *zai_read_property_direct_literal(zend_class_entry *ce, zend_object *obj, const char *name, int name_len);
+#define zai_read_property_direct_literal(obj, ce, name) zai_read_property_direct_literal(obj, ce, ZEND_STRL(name))
```

Comment:
What do you think about naming the function `zai_read_property_direct_cstr` and leaving the macro as `zai_read_property_direct_literal`? This way if I have a ptr + len pair already for some reason I can call the function directly without undef'ing the macro.

Reply:
I agree, I was already doubting that decision myself :-D